confluence-syntax-xhtml/src/main/java/org/xwiki/contrib/confluence/parser/xhtml/internal/wikimodel/URLTagHandler.java

```diff
@@ -53,6 +54,9 @@ protected void begin(TagContext context)
         WikiParameter urlParameter = context.getParams().getParameter("ri:value");
         if (urlParameter != null && container instanceof URLContainer) {
             ((URLContainer) container).setURL(urlParameter.getValue());
+        } else if (urlParameter != null && container instanceof ConfluenceMacro) {
```

Comment:
`urlParameter != null` is checked twice. Could you simplify? Maybe, If it's null, just skip it.